doc/fwutil/fwutil.md

```diff
@@ -462,6 +540,200 @@ Aborted!
+SSD firmware update is logged in a platform defined designated file and it will be updated during cold reboot.
+CPLD firmware update is going to be done with this command and the CPLD completion activity will be performed during cold reboot
+once all reboot processes are finished. The CPLD completion activity for this case is a power cycle triggered by the hw register setting.
+```bash
+root@sonic:~# fwutil auto-update fw --yes --image=next --boot=cold
+Available FW updates with fast-reboot from next boot image : SSD and CPLD
+SSD firmware update will be done during cold-reboot with version 5 (current version 4)
+...
+CPLD firmware update is in progress with version 5 (current version 4)
+...
+CPLD firmware update is completed with version 5 (current version 4)
+...
+All firmware update has been completed.
+```
+
+**Supported options:**
+3. -i|--image - update FW using current/next SONiC image
+4. -b|--boot - following boot option after the upgrade
+6. -z|--fw-image - firmware package downloaded during run time (this is an exclusive option from --image)
+
+**Note:** the default option is _--image=current_ and _--boot=any_
+
+##### 2.2.2.4.3 Platform Component Firmware Update Utility
+When Vendor doesn't have platform API ready to support all platform component APIs including the auto-update interface,
+the platform component utility will perform the equivalent process of `auto-update` platform component api.
+The component utility can perform the firmware update if the firmware update doesn't need any boot action required after the update.
+Otherwise, it will create a task file if any process or handling for the component firmware update needs to be done during the reboot
+and also if the update can be done for the specified reboot type.
+The componenet utility should be defined with key value `utility` in the component object of `platform_component.json` to be called by fwutil instead of the platform api.
+The task file will be platform-specific.
+
+**Example:**
+1. `platform_component.json` with platform specific utility.
+```json
+{
+    "component": {
+        "BIOS": {
+            "firmware": "bios.bin",
+            "version": "0ACLH003_02.02.010",
+        },
+        "CPLD": {
+            "firmware": "cpld.bin",
+            "utility": "cpldupdate",
+            "version": "10",
+        },
+        "SSD": {
+            "firmware": "SSD.bin",
+            "utility": "ssd_fw_update",
+            "version": "5",
+        }
+    }
+}
+```
+
+The Utility can have two passing arguments and one option for actions from fwutil.
+1.	Firmware
+2.	Boot action following after the firmware auto-update process
+3.	Fwutil action : -a(--autoupdate)
+**Utility can be supported for other platform api substitues like `compoenent_update` and `compoenent_install` with |-u(--update)|-i(--install)**
+
+The expected utility response can be as follows.
+```json
+"True" : "completed"
+"True" : "scheduled"
+"False" : Error
+```
```

Comment:
The "fwutil show auto-update" command has status as "updated but need a power-cycle". Please explain how these status will be derived from these utility return strings described above.